platform/files.go

```diff
@@ -102,6 +103,35 @@ func (t *TargetMetadata) IsSatisfiedBy(o *buildpack.TargetMetadata) bool {
 	return true
 }
 
+// IsValidRebaseTargetFor treats optional fields (ArchVariant and Distribution fields) as wildcards if empty, returns true if all populated fields match
+func (t *TargetMetadata) IsValidRebaseTargetFor(appTargetMetadata *TargetMetadata) bool {
+	if t.Arch != appTargetMetadata.Arch || t.OS != appTargetMetadata.OS {
+		return false
+	}
+	if t.ArchVariant != "" && appTargetMetadata.ArchVariant != "" && t.ArchVariant != appTargetMetadata.ArchVariant {
+		return false
+	}
+
+	if t.Distribution != nil && appTargetMetadata.Distribution != nil {
+		if t.Distribution.Name != appTargetMetadata.Distribution.Name {
+			return false
+		}
+		if t.Distribution.Version != appTargetMetadata.Distribution.Version {
+			return false
+		}
```

Comment:
Should we consider "missing" as "matches any" here? I actually don't think we do that in the detector logic, but maybe we should? cc @joe-kimmel-vmw

Reply:
Did you two get a chance to chat @natalieparellano?

Thinking through the transition from 0.11 to 0.12

Builds app on 0.11 - run image has no distro name/version
Run image gets updated with labels
Platform gets updated to 0.12
Rebase is executed
Fails due to "" on old not equal to "ubuntu/22.04" of new

Hmm. I think it would actually fail earlier as Arch != Arch or OS != OS.

Does that sound right as it is coded/spec'd? Do we want platforms to have added the labels during 0.11 including rebasing all images? Or should we potentially skip this bit of validation entirely if the target image was built with < 0.12?

Builds app on 0.11 - run image has no distro name/version
Run image gets updated with labels
Rebase image on 0.11 - OS/Arch/Distro/etc. validation is skipped
Successful Rebase
Platform gets updated to 0.12
Rebase is executed - validation occurs
Successful Rebase as everything matches from before

Reply:
> Or should we potentially skip this bit of validation entirely if the target image was built with < 0.12?

That makes sense to me. If the image was built on an older platform there is no way for the labels to be there.

Reply:
Took care of that in [bd6b6c6](https://github.com/buildpacks/lifecycle/pull/1056/commits/bd6b6c6879fa90f0b23503ac198eda553e7a4727)